File: erpnext/accounts/company_loans/company_loans.js
// Copyright (c) 2020, Frappe Technologies Pvt. Ltd. and contributors
// For license information, please see license.txt

frappe.ui.form.on('Company Loans', {
    
	  in:(frm)=>{
          update_total(frm);
	  /*	frappe.call({
	  		doc:frm.doc,
	  		method:'show_payment_shedule',
	  		callback :function(r){
	  			console.log(r.message)
	  		}
	  	})*/

	  	

      

  },



	
});

frappe.ui.form.on('benifits rates', 'rate', function(frm){
	

  update_total(frm);


    
});



function update_total(frm){
	
   	 var total =0;
	//console.log(d)
	//console.log(frm.doc.benifits)
     for(var i=0;i<frm.doc.benifits.length;i++){
     	if(frm.doc.benifits[i]["add_in_total"]==1){
     		total +=frm.doc.benifits[i]["rate"]|0;

     	}
     }
     if(frm.doc.in){
     	var profit=total*frm.doc.in;
     	var net=(profit/100)+frm.doc.in;
     	cur_frm.set_value("total", net);
     }
	 cur_frm.set_value("total_percent", total);
	 refresh_field("total_percent");
	 show_payment_shedule(frm)
}

function show_payment_shedule(frm){
	frm.clear_table("payment_shedules"); 
	frm.refresh_fields();
	// console.log(frm.doc)
	//frappe.throw(';test')
         frappe.call({
        'doc': frm.doc,
        'method': 'show_payment_shedule',
       
        callback: function(r) {
          console.log('r')
        }
    });
}

File: erpnext/accounts/company_loans/company_loans.py
# -*- coding: utf-8 -*-
# Copyright (c) 2020, Frappe Technologies Pvt. Ltd. and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe
from frappe.model.document import Document
import datetime
class CompanyLoans(Document):

	def show_payment_shedule(self):
		if self.no_of_days  and self.due_date  and self.the_number_of_payments_made and self.total :
			start_days=datetime.datetime.strptime(self.date,'%Y-%m-%d')
			date =datetime.datetime.strptime(self.due_date,'%Y-%m-%d')
			numofpyments=int(self.the_number_of_payments_made)
			total=float(self.total)
			amountscedules=float(total)/float(numofpyments)
			numofdays=int(self.no_of_days)
			for i in range(0,int(numofpyments)):
				date += datetime.timedelta(days=numofdays)
				row = self.append("payment_shedules", {})
				row.date=date
				row.amount=amountscedules

			return "success"
		return"failed"

		


        


		

